### ISSUE TITLE
System information error when error is fixed but page wasn't refreshed

### ISSUE THREAD

**Issue by snky1987** [magento/magento2#919]
When you open admin panel in few windows and you do the change in the layout then system throws System Message about invalidated cache.

Then, when you fix this issue in a new window and don't refresh the old one you will still see a bar about system error but when you click on it there is no message in a pop up.

I believe it would be worth to throw a message like "All system warning have been fixed. Please refresh the page" or similar. Just in case as empty message looks a bit suspicious.

Screen prove: http://prntscr.com/5p48py

**Comment by otoolec** [comment]
Thanks for reporting this bug.  I've opened up MAGETWO-32969 internally to track and resolve the issue.

**Comment by marcozheng** [comment]
Hi @snky1987, thank you for spending time and efforts trying to make Magento 2 better!

The M2 development team has reviewed this issue and evaluated the changes you are proposing.
We believe that this is a very important and necessary improvement we should add to M2.

There are a few suggestions we'd like to make before we merge your commits in:
* Severity level - instead using the value returned from $this->getRequest()->getParam('severity'), let's go ahead and set the severity to be the "notice" level for all cases. So basically, let's set (string)\Magento\Framework\Notification\MessageInterface::SEVERITY_NOTICE as the value for 'severity' for the default message.
* Default message - let's move the declaration of default message into the if (empty($result)) block, because it will be only used here - no need to declare the default message when the $result is not empty. Basically something like this:
```
        if (empty($result)) {
            $result[] = [
                'severity' => (string)\Magento\Framework\Notification\MessageInterface::SEVERITY_NOTICE,
                'text' => 'All recent issues have been fixed. '
                    .'Please refresh the screen for an update.',
            ];
        }
```
* Message content: let's use something like 'You have viewed and resolved all recent system notices. Please refresh the web page to clear the notice alert.' The reason for this is that the system notices may not always be "issues", they can also be "warning" or simple "notices".
* We have just refactored our copyright docblocks recently, so please revert your change there and take whatever the latest upstream has.
* Please revert the code style change you proposed for the following lines:
```
        $this->getResponse()->representJson(
            $this->_objectManager->get('Magento\Core\Helper\Data')->jsonEncode($result)
        );
```
* All the other code style changes look good and reasonable

Again, we really appreciate your time and efforts, and we are looking forward to accepting your contribution!

**Comment by anupdugar** [comment]
Also we encourage use of Dependency Injection instead of using objectManager
e.g.:
```
    /**
     * Initialize ListAction
     * 
     * @param \Magento\Backend\App\Action\Context $context
     * @param \Magento\Core\Helper\Data $coreHelper
     */
    public function __construct(\Magento\Backend\App\Action\Context $context, \Magento\Core\Helper\Data $coreHelper)
    {
        $this->coreHelper = $coreHelper;
        parent::__construct($context);
    }
```

and then

```
$this->getResponse()->representJson(
            $this->coreHelper->jsonEncode($result)
        );
```

**Comment by anupdugar** [comment]
@snky1987 Did you get a chance to look at the suggested code changes for your PR?

**Comment by snky1987** [comment]
@anupdugar Hi, I will today. Recently i had too much work with eBay service in AngularJS

**Comment by anupdugar** [comment]
No problem @snky1987. Feel free to get back if you need any inputs or have any suggestions or recommendations.

**Comment by snohe** [comment]
Hi @snky1987, we haven't heard back from you for quite some time.  To avoid closure of this issue, can you update us by end of this week?

**Comment by anupdugar** [comment]
Closing this issue since the PR is being tracked in #922.

**Comment by otoolec** [comment]
Re-opening this since #922 was closed.

**Comment by marcozheng** [comment]
The M2 dev team has already taken in the commits from @snky1987 and applied the recommended changes from the comments above. The internal ticket has been closed and merged into mainline codebase.

@otoolec @anupdugar I believe this ticket can now be closed.

**Comment by vpelipenko** [comment]
@snky1987, wait for update public repository. We will inform you when fix is available.

**Comment by snohe** [comment]
@snky1987, this is now resolved in 0.74.0-beta1.  Thank you again for submitting this issue.  Feel free to reach out to us should you have any questions.
